### PR TITLE
Softcode path

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | /
| Type?         | improvement
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | Everything should work exactly as before.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11647)

### CODE REVIEW

**Comment by MathiasReker** [comment]
__DIR__ is evaluated at compile-time, while dirname(__FILE__) means a function-call and is evaluated at execution-time so, __DIR__ is faster.

dir-vs-dirname-file

**Comment by MathiasReker** [comment]
Isn't the doc wrong? https://devdocs.prestashop.com/1.7/modules/creation/good-practices/

Shouldn't this `_PS_CONFIG_DIR_ . '/config.inc.php'` output `/../config//config.inc.php` with double forslash?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Really nice @MathiasReker 

For:
> _PS_CONFIG_DIR_ . '/config.inc.php' output /../config//config.inc.php

it's not a problem, adding a useless `/` is only a security to be sure if the const does not have a ending `/`

**Comment by MathiasReker** [comment]
@PierreRambaud Thanks for the explanation :)

**Comment by MathiasReker** [comment]
Travis CI though some errors about `_PS_CONFIG_DIR_`. Is it not possible to use this constant everywhere?

**Comment by MathiasReker** [comment]
I tried to fix travis CI, but I failed. I need some help. :)

File: src/PrestaShopBundle/Command/ModuleCommand.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ protected function init(InputInterface $input, OutputInterface $output)
         $this->translator = $this->getContainer()->get('translator');
         $this->input = $input;
         $this->output = $output;
-        require $this->getContainer()->get('kernel')->getRootDir() . '/../config/config.inc.php';
+        require(_PS_CONFIG_DIR_ . 'config.inc.php');
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Command/ModuleCommand.php]: You must run php-cs-fixer. You add parenthesis here, but require is not a function :) (even if it's work)

Comment (MathiasReker) [reply src/PrestaShopBundle/Command/ModuleCommand.php]: csfixer
I can use PHP-CS-Fixer manually from FriendsOfPHP with `php-cs-fixer.phar` , but I did never figure out how to use `php ./vendor/bin/php-cs-fixer fix` from inside PrestaShop. I am on windows 10. I did install composer. Can you learn me how to do that? For a lot of files, it is not easy to do it manually.

I attached a screenshot.

Comment (PierreRambaud) [reply src/PrestaShopBundle/Command/ModuleCommand.php]: I really don't know how Windows and its ecosystem works :/ Maybe try to call the script directly: ./vendor/friendsofphp/php-cs-fixer/php-cs-fixer ?

Comment (MathiasReker) [reply src/PrestaShopBundle/Command/ModuleCommand.php]: `./vendor/friendsofphp/php-cs-fixer/php-cs-fixer` worked! Thank you.

**Comment by MathiasReker** [comment]
PRESTASHOP_TEST_TYPE=e2e did not pass 😕

**Comment by mickaelandrieu** [comment]
> PRESTASHOP_TEST_TYPE=e2e did not pass

it's because `_PS_CONFIG_DIR_` is not always defined.

Ex:

```
[Mon Dec 10 02:01:54.507421 2018] [php7:warn] [pid 4511] [client 127.0.0.1:54840] PHP Warning:  Use of undefined constant _PS_CONFIG_DIR_ - assumed '_PS_CONFIG_DIR_' (this will throw an Error in a future version of PHP) in /home/dev/Projects/PrestaShop/admin-dev/index.php on line 42
```

Imho, you should remove this commit and address this improvement in another pull request :+1:

**Comment by MathiasReker** [comment]
I will remake this PR later.

**Pull request closed without merging**